llvm/test/CodeGen/DirectX/abs.ll

```diff
@@ -1,5 +1,5 @@
 ; RUN: opt -S  -dxil-intrinsic-expansion  < %s | FileCheck %s --check-prefixes=CHECK,EXPCHECK
-; RUN: opt -S  -dxil-op-lower  < %s | FileCheck %s --check-prefixes=CHECK,DOPCHECK
+; RUN: opt -S  -dxil-op-lower  -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s --check-prefixes=CHECK,DOPCHECK
```

Comment:
> (just checking) is "library" the right thing to use here, rather than "compute"? I don't think we're planning on support library shaders properly for a while and I wonder how stable the design is for anything involving them in clang.

I did consider specifying `compute`.

I noticed that tests in `clang/test/CodeGenHLSL/builtins/` have used the triple `dxil-pc-shadermodel6.3-library`. While it is not necessarily a very strong reason, I just chose to adapt the same string - particularly since the `EnvironmentType` is not consumed yet in the DXIL Op spec and/or lowering. I expect that to be part of changes in the near future - but can change it for now, if there is a strong push for it.

Reply:
Following the existing example seems reasonable to me.  It would probably be good to understand how that was chosen and to spread that knowledge a bit.  It looks like @llvm-beanz added some of those original tests that used library, maybe he can educate us.

Reply:
For the HLSL source tests we tend to use library shaders because they (1) don't require an entry point and (2) basically everything that is legal in all shader stages is legal in them.

That makes it easier for us to write tests that won't need updating as we implement more of the language and get stricter about enforcement.

Using the library triple on IR tests can in some situations provide a similar benefit, so it is probably the right approach.